pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -872,12 +898,39 @@ pub mod pallet {
 		pub fn set_season(
 			origin: OriginFor<T>,
 			season_id: SeasonId,
-			season: SeasonOf<T>,
+			season: Option<SeasonOf<T>>,
+			season_meta: Option<SeasonMeta>,
+			season_schedule: Option<SeasonScheduleOf<T>>,
+			trade_filters: Option<TradeFilters>,
 		) -> DispatchResult {
 			Self::ensure_organizer(origin)?;
-			let season = Self::ensure_season(&season_id, season)?;
-			Seasons::<T>::insert(season_id, &season);
-			Self::deposit_event(Event::UpdatedSeason { season_id, season });
+
+			if let Some(mut season_update) = season.clone() {
```

Comment:
With the `ref` keyword you can omit the `clone` as we only need a borrowed value afterward.